src/main/java/org/dsa/iot/alarm/AbstractAlarmObject.java

```
@@ -0,0 +1,322 @@
+     */
+    protected void doStart() {
+    }
+
+    /**
+     * Override point, called by the steady().  Started and steady will be true.
+     */
+    protected void doSteady() {
+    }
+
+    /**
+     * Override point, called by the stop().  Started and steady will be false.
+     */
+    protected void doStop() {
+    }
+
+    @Override public AlarmObject getChild(int index) {
+        return children.get(index);
+    }
+
+    /**
+     * The SDK node this object represents.
+     */
+    public Node getNode() {
+        return this.node;
+    }
+
+    @Override public AlarmObject getParent() {
+        return parent;
+    }
+
+    /**
+     * A convenience that first looks for the config, and if not found, tries for
+     * the attribute.
+     */
+    public Value getProperty(String name) {
+        Value ret = node.getConfig(name);
+        if (ret == null) {
+            ret = node.getRoConfig(name);
+        }
+        if (ret == null) {
+            ret = node.getAttribute(name);
+        }
+        return ret;
+    }
+
+    @Override public void init(Node node) {
+        if (node == null) {
+            throw new NullPointerException("SDK node cannot be null.");
+        }
+        if (this.node != null) {
+            throw new IllegalStateException("SDK node already set.");
+        }
+        this.node = node;
+        node.setRoConfig(JAVA_TYPE, new Value(getClass().getName()));
+        AlarmObject alarmObject;
+        if ((node != null) && node.getHasChildren()) {
+            Map<String,Node> kids = node.getChildren();
+            if (kids != null) {
+                for (Node kid : kids.values()) {
+                    alarmObject = AlarmUtil.tryCreateAlarmObject(kid);
+                    if (alarmObject != null) {
+                        addChild(alarmObject);
+                    }
+                }
+            }
+        } else {
+            System.out.println(getClass().getName());//XXX
```

Logging or something here? Maybe throw an error, I don't know.